### PR TITLE
Tell the user about --lib when new-installing a package with no exes

### PR DESCRIPTION
**Pull request by fgaz**
/cc @hvr

### CODE REVIEW

File: cabal-install/Distribution/Client/CmdInstall.hs [hunk 1]
```diff
@@ -614,8 +614,16 @@ symlinkBuiltPackage :: Verbosity
 symlinkBuiltPackage verbosity overwritePolicy
                     mkSourceBinDir destDir
                     (pkg, components) =
-  traverse_ symlinkAndWarn exes
+  if null exes
+    then warn verbosity $ "You asked to install executables, "
```
Comment (fgaz) [inline cabal-install/Distribution/Client/CmdInstall.hs]: `warn` or `die'`?

Comment (fgaz) [reply cabal-install/Distribution/Client/CmdInstall.hs]: I'm for `warn`, since technically cabal succeeds in installing "all the exes of the package"

**Comment by 23Skidoo** [comment]
Huh, 100% green.

**Comment by fgaz** [comment]
I think 5bc88cd3ea97e3888d81dd9d125db34001b8cd94 fixed it

**Comment by fgaz** [comment]
Cherry-picker to 2.4

**Comment by 23Skidoo** [comment]
@fgaz Oh BTW, when you cherry-pick patches, please use `git cherry-pick -x`.

> I think 5bc88cd3ea97e3888d81dd9d125db34001b8cd94 fixed it

Hmm, how? Bootstrap configuration [was green before](https://travis-ci.org/haskell/cabal/builds/451128261?utm_source=github_status&utm_medium=notification). I think it was the release of 8.6.2 final (that configuration was failing), but I'm not sure what exactly changed.

**Comment by fgaz** [comment]
> when you cherry-pick patches, please use git cherry-pick -x
oh,sorry. I'll do that from now on.
> the release of 8.6.2 final 
that's probably it then

**Pull request merged**